test/functional/tests/security/test_security_user.py

```diff
@@ -0,0 +1,468 @@
+user_casadm_bin_dest_path = "/bin/casadm"
+ioclass_config_path = "/etc/opencas/ioclass-config.csv"
+ioclass_config_copy_path = "/etc/opencas/ioclass-config-copy.csv"
+user_name = "user"
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_user_cli():
+    """
+        title: Test that OpenCAS do not allow to change parameters in CLI by non-root user.
+        description: |
+          Checking if changing parameters in CLI by non-root user is forbidden by OpenCAS,
+          but is permitted with 'sudo' command.
+        pass_criteria:
+          - Non-root user can only print help and CAS version.
+          - Sudoer user is allowed to change OpenCAS parameters in CLI with sudo.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start OpenCAS."):
+        cache = casadm.start_cache(cache_dev, force=True)
+
+    with TestRun.step("Add core to cache and mount it."):
+        core_dev.create_filesystem(Filesystem.ext3)
+        core = cache.add_core(core_dev)
+        core.mount(mount_point)
+
+    with TestRun.step("Create 'opencas.conf' from running configuration."):
+        init_config.create_init_config_from_running_configuration()
+
+    with TestRun.step(f"Copy casadm bin from {system_casadm_bin_path} "
+                      f"to {user_casadm_bin_dest_path}."):
+        casadm_bin = fs_utils.parse_ls_output(fs_utils.ls_item(f"{system_casadm_bin_path}"))[0]
+        casadm_bin_copy = casadm_bin.copy(user_casadm_bin_dest_path, True)
+        casadm_bin_copy.chmod_numerical(777)
+
+    with TestRun.step("Copy IO class config."):
+        io_conf = fs_utils.parse_ls_output(fs_utils.ls_item(f"{ioclass_config_path}"))[0]
+        io_conf_copy = io_conf.copy(ioclass_config_copy_path, force=True)
+
+    with TestRun.step("Unmount core."):
+        core.unmount()
+
+    with TestRun.step("Stop OpenCAS."):
+        casadm.stop_all_caches()
+
+    with TestRun.step("Add non-root user account."):
+        TestRun.executor.run(f"useradd -N -r -l {user_name}")
+        user_home_dir = fs_utils.parse_ls_output(fs_utils.ls_item(f"/home/{user_name}"))[0]
+        user_home_dir.chmod_numerical(777, True)
+
+    with TestRun.step("Try to start OpenCAS."):
+        try:
+            output = run_as_other_user(cli.start_cmd(cache_dev.system_path), user_name)
+            if output.exit_code == 0:
+                TestRun.fail("Starting OpenCAS should fail!")
+        except CmdException:
+            TestRun.LOGGER.info("Non-root user cannot start OpenCAS.")
+
+    with TestRun.step("Try to stop OpenCAS."):
```

Comment:
General:
Wouldn't it be better to start a cache before trying to perform operations that require it? Right now I think that even if we had a bug and unauthorized user could stop cache/add core/etc we wouldn't know because those operation will fail anyways (cache is not there).

Reply:
You're right.